OneBusAway/ui/stops/OBAStopViewController.m

```diff
@@ -363,9 +363,19 @@ - (void)deleteAlarmForArrivalAndDeparture:(OBAArrivalAndDepartureV2*)dep {
     NSURLRequest *request = [self.modelService.obaJsonDataSource requestWithURL:alarm.alarmURL HTTPMethod:@"DELETE"];
     [self.modelService.obaJsonDataSource performRequest:request completionBlock:^(id responseData, NSUInteger responseCode, NSError *error) {
         [self.modelDAO removeAlarmWithKey:dep.alarmKey];
+        // Replace with a new row and update it.
+        [self updateAlarmStateForRowAtIndexPath:[self indexPathForModel:dep]];
     }];
 }
 
+- (void)updateAlarmStateForRowAtIndexPath:(NSIndexPath *)indexPath {
+    OBADepartureRow *row = (OBADepartureRow *)[self rowAtIndexPath:indexPath];
+    OBADepartureRow *newRow = row;
```

Comment:
To not reload the entire table view, I decided to reload just a single table cell.
And I create a new row that is the same as the old one but with the new updated alarm state.
Then replace the old row and reload it.